### DISCUSSION TITLE
The sassVariables configuration of @quasar/vite-plugin is incompatible with the configuration of vite's preprocessorOptions

### DISCUSSION THREAD

**Discussion by zouyaoji** [quasarframework/quasar#12950] [General - Components / Directives / etc] [answered]
I originally configured my own scss variables imported via @quasar/vite-plugin's sassVariables config, but found a lot of duplicate css. I suspect that the problem is caused by sassVariables, so I want to introduce my own scss variables through the configuration of vite itself (https://vitejs.dev/config/#css-preprocessoroptions), but I found that it does not take effect, but once I Annotate the @quasar/vite-plugin plugin and it will take effect.

Therefore, I would like to ask why the sassVariables configuration of @quasar/vite-plugin and the configuration preprocessoroptions of vite itself cannot be used at the same time?

**Comment by rstoenescu** [comment, marked as answer]
Hi,

Just add your own scss variables in the variables file indicated by our Vite plugin. As long as it's only variables -- and not code that outputs real CSS, you're gonna be fine. What I suspect that you are doing is the latter (either directly or by importing other files that do that).
I wonder what makes you think that the variables file is only for Quasar ones. It's a SCSS/Sass file after all, we're not reinventing the wheel or writing a special parser to error out on non-Quasar variables. Any SCSS variable will do :)

As for why the preprocessor options do not take effect -> this is due to the architecture of Vite where plugins are applied after your vite.config.js file is read (which is fine). But this is besides the point, considering what you want to achieve and what I've written above.

**Reply by zouyaoji** [reply]
You're right, I wrote some code that outputs real CSS, for the multi-theme switching feature. Was migrated from webpack, but I found that under vite, it's not perfect because I found a lot of duplicate css.

This is my test project: https://github.com/zouyaoji/vue-cesium-demo and test address: https://zouyaoji.top/vue-cesium-demo/index.

The core idea is to define multiple sets of variables in advance, and then change the theme-related classes on the body when switching themes to achieve the purpose of switching themes. This doesn't seem to be able to separate css and variables?

Maybe this logic can't be solved perfectly in vite, so I would like to ask you how to easily realize the function of switching between multi-theme styles in the quasar+vite project. Can you give me some suggestions or references?
